pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1071,6 +1075,21 @@ private void closeStreamConsumers() {
     if (_acquiredConsumerSemaphore.compareAndSet(true, false)) {
       _partitionGroupConsumerSemaphore.release();
     }
+    if (_tableConfig.getUpsertConfig().isEnableSnapshot()) {
```

Comment:
Why do we take snapshot here? `_isReadyToConsumeData` won't be picked up by the next segment.
I feel we can take the snapshot just after acquiring the consumer semaphore

Reply:
@Jackie-Jiang thanks for review. When acquiring the consumer semaphore, does it guarantee the previous segment is sealed?

I plan to acquire all segments and persist snapshot for all sealed segments, is it possible that the last segment is not sealed yet?